common/types.ts

```diff
@@ -13,3 +14,25 @@ export interface FetchDataFrameContext {
 }
 
 export type FetchFunction<T, P = void> = (params?: P) => Observable<T>;
+
+// ref: https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/job-states.html
+export enum SparkJobState {
```

Comment:
nit: I think we can take out the ref to spark and the documentation link since these could be applied to things that aren't Spark related.